frontend/src/libs/components/dropdown/dropdown.tsx

```diff
@@ -99,20 +71,36 @@ const Dropdown = <T extends FieldValues>({
     setIsMenuOpen(false);
   }, []);
 
-  const stylesConfig = useMemo(() => getStyles(isMenuOpen), [isMenuOpen]);
+  const classNamesConfig = useMemo(
+    () => getClassNames(isMenuOpen),
+    [isMenuOpen],
+  );
+
+  const { field } = control
+    ? // eslint-disable-next-line react-hooks/rules-of-hooks
+      useFormController({
+        name: name as FieldPath<T>,
+        control,
+      })
+    : { field: null };
```

Comment:
Not good, this rule must be enabled

Reply:
Is this conditional hook application? This is no-no.

Reply:
doesn't work without it